src/client/error.rs

```diff
@@ -0,0 +1,264 @@
+use bson;
+use byteorder;
+
+use client::coll::options::WriteModel;
+use std::{error, fmt, io, result, sync};
+
+/// A type for results generated by MongoDB related functions, where the Err type is mongodb::Error.
+pub type Result<T> = result::Result<T, Error>;
+
+/// The error type for MongoDB operations.
+#[derive(Debug)]
+pub enum Error {
```

Comment:
Put comments about what each error type means. It's not clear what `CursorMissingError` should represent, for example.